### PR TITLE
[TypeSpec Validation] Disable simple-git debugging

### PR DESCRIPTION
**Pull request by mikeharder**

### CODE REVIEW

File: eng/tools/TypeSpecValidation/src/rules/git-diff.ts [hunk 1]
```diff
@@ -1,16 +1,14 @@
...
 
-debug.enable("simple-git");
 export class GitDiffRule implements Rule {
   readonly name = "GitDiff";
   readonly description = "Checks if previous rules resulted in a git diff";
 
   async execute(): Promise<RuleResult> {
     const git = simpleGit();
-    let gitStatusIsClean = await (await git.status(["--porcelain"])).isClean();
+    let gitStatusIsClean = (await git.status(["--porcelain"])).isClean();
```
Comment (mikeharder) [inline eng/tools/TypeSpecValidation/src/rules/git-diff.ts:11]: Removed unecessary `await`

**Review by weshaggard — APPROVED** [review]

**Comment by ckairen** [comment]
why do we not want the simple git debug messages?

**Comment by mikeharder** [comment]
> why do we not want the simple git debug messages?

I think it's too noisy to have enabled by default.  I think it will confuse spec authors more than it will help them.  We could add a `--verbose` flag to `tsv` if we wanted to enable it optionally.  

For example, I think this:

```
Executing rule: GitDiff
Rule GitDiff failed
{"not_added":[],"conflicted":[],"created":[],"deleted":[],"modified":["specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-08-01-preview/examples/extensions_chat_completions.json"],"renamed":[],"files":[{"path":"specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-08-01-preview/examples/extensions_chat_completions.json","index":" ","working_dir":"M"}],"staged":[],"ahead":0,"behind":0,"current":"HEAD","tracking":null,"detached":true}diff --git a/specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-08-01-preview/examples/extensions_chat_completions.json b/specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-08-01-preview/examples/extensions_chat_completions.json
index 12ef3a0942..8b71198133 100644
--- a/specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-08-01-preview/examples/extensions_chat_completions.json
+++ b/specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-08-01-preview/examples/extensions_chat_completions.json
@@ -18,7 +18,7 @@
       ],
       "dataSources": [
         {
-          "type": "AzureCognitiveSearch",
+          "type": "azureChatExtensionType",
           "parameters": {
             "source_specific_parameter_1": "foo",
             "source_specific_parameter_2": {
```

Is clearer than this:

```
Executing rule: GitDiff
2023-08-23T10:46:23.955Z simple-git [GitExecutor] [SPAWN] git [ 'status', '--porcelain', '-b', '-u', '--null', '--porcelain' ]
2023-08-23T10:46:24.389Z simple-git [GitExecutor] [HANDLE] retrieving task output complete
2023-08-23T10:46:24.390Z simple-git [GitExecutor] [SPAWN] git [ 'status', '--porcelain', '-b', '-u', '--null' ]
2023-08-23T10:46:24.704Z simple-git [GitExecutor] [HANDLE] retrieving task output complete
2023-08-23T10:46:24.704Z simple-git [GitExecutor] [SPAWN] git [ 'diff' ]
2023-08-23T10:46:24.822Z simple-git [GitExecutor] [HANDLE] retrieving task output complete
Rule GitDiff failed
{"not_added":[],"conflicted":[],"created":[],"deleted":[],"modified":["specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-08-01-preview/examples/extensions_chat_completions.json"],"renamed":[],"files":[{"path":"specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-08-01-preview/examples/extensions_chat_completions.json","index":" ","working_dir":"M"}],"staged":[],"ahead":0,"behind":0,"current":"HEAD","tracking":null,"detached":true}diff --git a/specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-08-01-preview/examples/extensions_chat_completions.json b/specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-08-01-preview/examples/extensions_chat_completions.json
index 12ef3a0942..8b71198133 100644
--- a/specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-08-01-preview/examples/extensions_chat_completions.json
+++ b/specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2023-08-01-preview/examples/extensions_chat_completions.json
@@ -18,7 +18,7 @@
       ],
       "dataSources": [
         {
-          "type": "AzureCognitiveSearch",
+          "type": "azureChatExtensionType",
           "parameters": {
             "source_specific_parameter_1": "foo",
             "source_specific_parameter_2": {
```

Looking at the top output, I think there is still more room for improvement, but disabling `simple-git` debugging is one step in the right direction.

**Review by ckairen — APPROVED** [review]

**Comment by mikeharder** [comment]
We may decide to re-enable this optionally in this PR: https://github.com/Azure/azure-rest-api-specs/pull/24934

**Pull request merged**